### ISSUE TITLE
Value callbacks always triggered with null first

### ISSUE THREAD

**Issue by ginovva320** [urish/firebase-server#2]
If I listen for the *value* event on a node, the snapshot that's returned contains no data on the first event. So if I'm using *once*, the value is null. But if I use *on*, the callback is called twice, once with null and once again with the data.

```javascript
var FirebaseServer = require('firebase-server');

var s = new FirebaseServer(5000, 'dummy.firebaseio.test', {
    states: {
        CA: 'California',
        AL: 'Alabama',
        KY: 'Kentucky'
    }
});

var Firebase = require('firebase');

var client = new Firebase('ws://dummy.firebaseio.test:5000');

client.child('states').child('CA').on('value', function(snap) {
    console.log('Value: ' + snap.val());
});
```

Output:
```
Value: null
Value: California
```

**Comment by urish** [comment]
Thanks for reporting, I could successfully reproduce the issue. Any idea what is causing this?

**Comment by ginovva320** [comment]
I don't know much about Firebase's internals, but my guess would be that the initial null is what's stored locally before a sync occurs. This sort of thing happens during a transaction normally, but in that case, the SDK "knows" to run the transaction function again until data is synced with the server.

**Comment by urish** [comment]
Just pushed a fix - can you please try it and let me know if it solves the problems? 

Once the solution will be verified, I will release it as 0.2.0. 

Thanks!

**Comment by ginovva320** [comment]
The value callbacks are working as expected now. Thanks for the quick fix!

**Comment by urish** [comment]
Great news! Released to npm as v0.2.0
